var/spack/repos/builtin/packages/lmod/package.py

```diff
@@ -46,6 +46,9 @@ class Lmod(AutotoolsPackage):
     depends_on('lua-luafilesystem', type=('build', 'run'))
     depends_on('tcl', type=('build', 'link', 'run'))
 
+    variant('auto-swap', default=False, description='Enable auto swapping conflicting modules')
```

Comment:
This isn't yet standardized, but I would like to standardize on underscores, not dashes. Dashes are used to disable variants, so `-auto-swap` can be confusing for humans. It's unclear whether it's the same as `~auto~swap` or `~auto-swap`.

Reply:
In a case like this I would consider using a single-valued variant which can take e.g. 2 values:
```suggestion
    variant('conflict', default='error', values=('error', 'swap'), multi=False, 
            description='Enable auto swapping conflicting modules')
```
This will be clearer in my opinion when specifying the spec on the command line since:
```
lmod conflict=swap
```
seems more explicit on the behavior than:
```
lmod +auto_swap
```
It also solves the underscore vs. dash issue by using a key-value pair. This is just a suggestion not a request, so feel free to drop it if you or @adamjstewart think it's better to proceed otherwise.

Reply:
I'd be open to making that change (and certainly removing the dash either way), but I'm slightly concerned that if the option is named something other than what a user might be expecting to look for, it could be overlooked by some users. For example, a new user to lmod might be looking for an auto_swap option without realize that conflict means the same thing. Maybe I'm overthinking it though so if thats not a real concern I can switch it to a key value pair.

Reply:
I personally prefer boolean variants, not multi-valued variants, especially when there are only two options. I agree with naming the option the same as the configure/cmake flag. I personally vote for `+auto_swap`, @alalazo?

Reply:
Boolean variant is fine with me. I just wanted to point out that also in the case of only two values one can use single-valued variants if:
- The labels used are more informative to the user than `True` and `False`
- It is foreseen that additional behavior might be added later